core/widgetdiv.js

```diff
@@ -85,18 +85,30 @@ Blockly.WidgetDiv.show = function(newOwner, rtl, dispose) {
 };
 
 /**
- * Destroy the widget and hide the div.
+ * Destroy the widget, reset the css, and hide the div.
  */
 Blockly.WidgetDiv.hide = function() {
-  if (Blockly.WidgetDiv.owner_) {
-    Blockly.WidgetDiv.owner_ = null;
-    Blockly.WidgetDiv.DIV.style.display = 'none';
-    Blockly.WidgetDiv.DIV.style.left = '';
-    Blockly.WidgetDiv.DIV.style.top = '';
-    Blockly.WidgetDiv.dispose_ && Blockly.WidgetDiv.dispose_();
-    Blockly.WidgetDiv.dispose_ = null;
-    Blockly.WidgetDiv.DIV.innerHTML = '';
+  if (!this.isVisible()) {
+    return;
   }
+
+  Blockly.WidgetDiv.owner_ = null;
+  Blockly.WidgetDiv.DIV.innerHTML = '';
+  Blockly.WidgetDiv.resetCss_();
+  Blockly.WidgetDiv.dispose_ && Blockly.WidgetDiv.dispose_();
+  Blockly.WidgetDiv.dispose_ = null;
+};
+
+/**
+ * Resets the css of the div.
+ * @private
+ */
+Blockly.WidgetDiv.resetCss_ = function() {
+  var style = Blockly.WidgetDiv.DIV.style;
+  style.cssText = '';
```

Comment:
I don't like this method of clearing the CSS. Best if we only add and remove what we care about.
That way if a developer really wants to add style elements to the div, we don't interfere with those.

Reply:
Wouldn't it be easier to just let the text input field clean up then?

Reply:
Yeah I see your point. Following on my previous comment, I would say let each field clean up anything it added to the dropdown div's style.

Reply:
Sounds good, I'll revert the previous commit.